Tests/AWSSDKSwiftCoreTests/AWSClientTests.swift

```diff
@@ -78,62 +78,40 @@ class AWSClientTests: XCTestCase {
 
 
     func testGetCredential() {
-        let client = createAWSClient(accessKeyId: "key", secretAccessKey: "secret")
+        let client = AWSClient(accessKeyId: "key", secretAccessKey: "secret", httpClientProvider: .createNew)
 
-        do {
-            let credentialForSignature = try client.credentialProvider.getCredential(on: client.eventLoopGroup.next()).wait()
-            XCTAssertEqual(credentialForSignature.accessKeyId, "key")
-            XCTAssertEqual(credentialForSignature.secretAccessKey, "secret")
-        } catch {
-            XCTFail(error.localizedDescription)
-        }
-    }
-
-    // this test only really works on Linux as it requires the MetaDataService. On mac it will just pass automatically
-    func testExpiredCredential() {
-        let client = createAWSClient()
-
-        do {
-            let credentials = try client.credentialProvider.getCredential(on: client.eventLoopGroup.next()).wait()
-            print(credentials)
-        } catch NIO.ChannelError.connectTimeout(_) {
-            // credentials request should fail. One possible error is a connectTimerout
-        } catch is NIOConnectionError {
-                // credentials request should fail. One possible error is a connection error
-//        } catch MetaDataServiceError.couldNotGetInstanceRoleName {
-            // credentials request fails in a slightly different way if it finds the IP
-        } catch {
-            XCTFail("Unexpected error \(error)")
-        }
+        var credential: Credential?
+        XCTAssertNoThrow(credential = try client.credentialProvider.getCredential(on: client.eventLoopGroup.next()).wait())
+        XCTAssertEqual(credential?.accessKeyId, "key")
+        XCTAssertEqual(credential?.secretAccessKey, "secret")
     }
 
     func testPartitionEndpoints() throws {
-        let client = createAWSClient(
+        let config = createServiceConfig(
             serviceEndpoints: ["aws-global":"service.aws.amazon.com"],
             partitionEndpoints: [.aws: (endpoint: "aws-global", region: .euwest1)]
         )
-        XCTAssertEqual(client.region, .euwest1)
-
-        let awsRequest = try client.createAWSRequest(operation: "test", path: "/", httpMethod: "GET")
+        XCTAssertEqual(config.region, .euwest1)
+        
+        let awsRequest = try AWSRequest(operation: "test", path: "/", httpMethod: "GET", configuration: config)
         XCTAssertEqual(awsRequest.url.absoluteString, "https://service.aws.amazon.com/")
     }
 
+    
     func testCreateAwsRequestWithKeywordInHeader() {
```

Comment:
Is it worth moving all the `AWSRequest` test into a new `AWSRequestTests` given these aren't related to `AWSClient` anymore